cli/azd/cmd/provision.go

```diff
@@ -101,51 +97,66 @@ func newProvisionAction(
 	formatter output.Formatter,
 	writer io.Writer,
 	commandRunner exec.CommandRunner,
+	middlewareRunner middleware.MiddlewareContext,
+	packageActionInitializer actions.ActionInitializer[*packageAction],
 ) actions.Action {
 	return &provisionAction{
-		flags:           flags,
-		accountManager:  accountManager,
-		projectManager:  projectManager,
-		resourceManager: resourceManager,
-		azdCtx:          azdCtx,
-		azCli:           azCli,
-		env:             env,
-		formatter:       formatter,
-		projectConfig:   projectConfig,
-		writer:          writer,
-		console:         console,
-		commandRunner:   commandRunner,
+		flags:                    flags,
+		accountManager:           accountManager,
+		projectManager:           projectManager,
+		resourceManager:          resourceManager,
+		azdCtx:                   azdCtx,
+		azCli:                    azCli,
+		env:                      env,
+		formatter:                formatter,
+		projectConfig:            projectConfig,
+		writer:                   writer,
+		console:                  console,
+		commandRunner:            commandRunner,
+		middlewareRunner:         middlewareRunner,
+		packageActionInitializer: packageActionInitializer,
 	}
 }
 
-func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	if p.flags.noProgress {
+func (pa *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	if pa.flags.noProgress {
 		fmt.Fprintf(
-			p.console.Handles().Stderr,
+			pa.console.Handles().Stderr,
 			//nolint:Lll
 			output.WithWarningFormat("--no-progress flag is deprecated and will be removed in the future.")+"\n")
 	}
 
+	packageAction, err := pa.packageActionInitializer()
+	if err != nil {
+		return nil, err
+	}
+
+	packageOptions := &middleware.Options{CommandPath: "package"}
+	_, err = pa.middlewareRunner.RunChildAction(ctx, packageOptions, packageAction)
```

Comment:
I was thinking that we wanted to add `package` before `provision` in the `up` implicit workflow, and not necessarily modify `provision`. Let's check with the team on this.